### PR TITLE
Support Vendored frameworks in the CocoaPods development pod

### LINKED ISSUE #134: Get caches failed after changed project directory
**My integration setup**
- CocoaPods cocoapods-xcremotecache plugin
- Contain 97 dependencies, including Development Pods. 

**Problem & Expectation**
- Local machine: cache hit rate is 80/97. But after moved to another directory, the cache hit rate is 40/97.

- CI machine, each build job has a different directory and the cache hit rate is 40/100.

Expectation: 
1. Change the source directory path should not affect the cache hit rate. 
2. CI jobs with different directory will hit the same cache as local machine.

**Minimal reproduction of the problem with instructions**
To reproduce this locally, we can test by this steps:
1. Provide producer then run consumer mode.
2. Count the total number of cached.
3. Move your project to another directory.
4. Delete Pod folder and rerun consumer mode.
5. Count the total number of consumer mode and compare.

**Environment**
* **cocoapods-xcremotecache:**  0.0.9
* **Xcode:** 13.2.1

### Notes
I've read `Build artifacts portability` in [Spotify blog](https://engineering.atspotify.com/2021/11/introducing-xcremotecache-the-ios-remote-caching-tool-that-cut-our-clean-build-times-by-70/) and understand XCRemoteCache already support `fingerprint_override_extension` & `debug-prefix-map`. 
Not sure why this not working for our case, it blocking us to apply XCRemoteCache to CICD system.

### PR DESCRIPTION
**Pull request by polac24**
If a development pod uses vendored_frameworks, the precompiled .framework is referenced from the `{RootProject}/Modules`, while all development Pods' `SRCROOT` are placed in `{RootProject}/Pods`. As a result, all files referenced dependencies from `Modules` use absolute paths (because that path is outside of `SRCROOT` or any other observed paths).

This PR allows referencing any file from the `{RootProject}` - both in the main project or Pods project. 

Impl details: sometimes we have to set `./../` (for Pods projects) and `./` (the main one) but fortunately, we already have logic for that because we reference xcprebuild, xcpostbuild etc. apps via `{RootProject}`. An idea is that all Xcode projects contain a build setting `XCRC_COOCAPODS_ROOT` that points to the `{RootProject}`. That ENV is always added to the `custom_rewrite_envs` so XCRemoteCahce knows to replace that path into a generic location: 

 #### Pods project
Screenshot 2022-05-17 at 19 58 53
#### Main project 
Screenshot 2022-05-17 at 20 00 05

#### How to test:
Sample project: https://github.com/canhth/nestedCocoapodsSample

name1's meta before:
```
{
  "inputs" : [

  ],
  "generationCommit" : "7417971ddde58a792d216a6468a1e0368f05ff74",
  "pluginsKeys" : {

  },
  "targetName" : "name1",
  "platform" : "iphoneos",
  "configuration" : "Debug",
  "rawFingerprint" : "952798faf4b055f19d97fec1c7fa5dc8",
  "fileKey" : "35d9d517b94bec38813884af68ab2173",
  "dependencies" : [
    "$(BUILD_DIR)\/Debug-iphoneos\/FaceDetection\/FaceDetection.framework\/Modules\/module.modulemap",
    "$(BUILD_DIR)\/Debug-iphoneos\/name2\/name2.framework\/Modules\/module.modulemap",
    "$(BUILD_DIR)\/Debug-iphoneos\/FaceDetection\/FaceDetection.framework\/Headers\/FaceDetection-Swift.h",
    "$(BUILD_DIR)\/Debug-iphoneos\/FaceDetection\/FaceDetection.framework\/Modules\/FaceDetection.swiftmodule\/arm64-apple-ios.swiftmodule.md5",
    "$(BUILD_DIR)\/Debug-iphoneos\/name2\/name2.framework\/Headers\/name2-Swift.h",
    "$(BUILD_DIR)\/Debug-iphoneos\/name2\/name2.framework\/Modules\/name2.swiftmodule\/arm64-apple-ios.swiftmodule.md5",
    "/Some/Absolute/path\/Modules\/FaceDetection\/FaceDetection\/Frameworks\/KYCQualityDetection.framework\/Headers\/GojekCV.h",
    "/Some/Absolute/path\/Modules\/FaceDetection\/FaceDetection\/Frameworks\/KYCQualityDetection.framework\/Headers\/HelperClasses.h",
    "/Some/Absolute/path\/Modules\/FaceDetection\/FaceDetection\/Frameworks\/KYCQualityDetection.framework\/Modules\/module.modulemap",
    "/Some/Absolute/path\/Modules\/name1\/name1\/Classes\/ReplaceMe.swift",
    "$(BUILD_DIR)\/Debug-iphoneos\/FaceDetection\/FaceDetection.framework\/Headers\/FaceDetection-umbrella.h",
    "$(SRCROOT)\/Target Support Files\/name1\/name1-dummy.m",
    "$(SRCROOT)\/Target Support Files\/name1\/name1-prefix.pch",
    "$(BUILD_DIR)\/Debug-iphoneos\/name1\/name1.framework\/Headers\/name1-umbrella.h",
    "$(BUILD_DIR)\/Debug-iphoneos\/name2\/name2.framework\/Headers\/name2-umbrella.h"
  ],
  "xcode" : "13C100"
}
``` 
name1's meta after: 
```
{
  "inputs" : [

  ],
  "generationCommit" : "7417971ddde58a792d216a6468a1e0368f05ff74",
  "pluginsKeys" : {

  },
  "targetName" : "name1",
  "platform" : "iphoneos",
  "configuration" : "Debug",
  "rawFingerprint" : "952798faf4b055f19d97fec1c7fa5dc8",
  "fileKey" : "35d9d517b94bec38813884af68ab2173",
  "dependencies" : [
    "$(BUILD_DIR)\/Debug-iphoneos\/FaceDetection\/FaceDetection.framework\/Modules\/module.modulemap",
    "$(BUILD_DIR)\/Debug-iphoneos\/name2\/name2.framework\/Modules\/module.modulemap",
    "$(BUILD_DIR)\/Debug-iphoneos\/FaceDetection\/FaceDetection.framework\/Headers\/FaceDetection-Swift.h",
    "$(BUILD_DIR)\/Debug-iphoneos\/FaceDetection\/FaceDetection.framework\/Modules\/FaceDetection.swiftmodule\/arm64-apple-ios.swiftmodule.md5",
    "$(BUILD_DIR)\/Debug-iphoneos\/name2\/name2.framework\/Headers\/name2-Swift.h",
    "$(BUILD_DIR)\/Debug-iphoneos\/name2\/name2.framework\/Modules\/name2.swiftmodule\/arm64-apple-ios.swiftmodule.md5",
    "$(XCRC_COOCAPODS_ROOT)\/Modules\/FaceDetection\/FaceDetection\/Frameworks\/KYCQualityDetection.framework\/Headers\/GojekCV.h",
    "$(XCRC_COOCAPODS_ROOT)\/Modules\/FaceDetection\/FaceDetection\/Frameworks\/KYCQualityDetection.framework\/Headers\/HelperClasses.h",
    "$(XCRC_COOCAPODS_ROOT)\/Modules\/FaceDetection\/FaceDetection\/Frameworks\/KYCQualityDetection.framework\/Modules\/module.modulemap",
    "$(XCRC_COOCAPODS_ROOT)\/Modules\/name1\/name1\/Classes\/ReplaceMe.swift",
    "$(BUILD_DIR)\/Debug-iphoneos\/FaceDetection\/FaceDetection.framework\/Headers\/FaceDetection-umbrella.h",
    "$(SRCROOT)\/Target Support Files\/name1\/name1-dummy.m",
    "$(SRCROOT)\/Target Support Files\/name1\/name1-prefix.pch",
    "$(BUILD_DIR)\/Debug-iphoneos\/name1\/name1.framework\/Headers\/name1-umbrella.h",
    "$(BUILD_DIR)\/Debug-iphoneos\/name2\/name2.framework\/Headers\/name2-umbrella.h"
  ],
  "xcode" : "13C100"
}
```

Fixes #134

### CODE REVIEW

File: cocoapods-plugin/README.md
Comment (polac24) [inline cocoapods-plugin/README.md:69]: Fixed in #57

File: cocoapods-plugin/lib/cocoapods-xcremotecache/command/hooks.rb
Comment (polac24) [inline cocoapods-plugin/lib/cocoapods-xcremotecache/command/hooks.rb:32]: It was a silly mistake in #52: there is `! CUSTOM_CONFIGURATION_KEYS ` in the expression: 
https://github.com/spotify/XCRemoteCache/blob/91505a59d2a3b8db3dc0eef90c6fbdc051141169/cocoapods-plugin/lib/cocoapods-xcremotecache/command/hooks.rb#L96

**Comment by canhth** [comment]
Hi @polac24,
Thank you so much for support this case.  💯 

Based on my understanding, the `XCRC_COOCAPODS_ROOT` is added to default `custom_rewrite_envs`, so... we don't have to add any other configuration, just to confirm.
Hope we can release the new version ASAP.

**Comment by polac24** [comment]
@canhth, that is correct. This PR adds `XCRC_COOCAPODS_ROOT` to all CocoaPods-generated projects so no extra steps are required from you.

Sure, let me release a new version once I get approval here.

**Review by PatrikBillgren — APPROVED** [review]

**Pull request merged**
